// angular2-grid/directives/NgGrid.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from 'angular2-grid/directives/NgGrid';
import * as import1 from '@angular/core/src/change_detection/change_detection_util';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
export class Wrapper_NgGrid {
  /*private*/ _eventHandler:Function;
  context:import0.NgGrid;
  /*private*/ _changed:boolean;
  /*private*/ _expr_0:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  subscription3:any;
  subscription4:any;
  subscription5:any;
  subscription6:any;
  constructor(p0:any,p1:any,p2:any,p3:any,p4:any) {
    this._changed = false;
    this.context = new import0.NgGrid(p0,p1,p2,p3,p4);
    this._expr_0 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
    (this.subscription3 && this.subscription3.unsubscribe());
    (this.subscription4 && this.subscription4.unsubscribe());
    (this.subscription5 && this.subscription5.unsubscribe());
    (this.subscription6 && this.subscription6.unsubscribe());
  }
  check_config(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.config = currValue;
      this._expr_0 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) {
      if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); }
      this.context.ngDoCheck();
    }
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    if ((eventName == 'mousedown')) {
      const pd_sub_0:any = ((<any>this.context.mouseDownEventHandler($event)) !== false);
      result = (pd_sub_0 && result);
    }
    if ((eventName == 'mousemove')) {
      const pd_sub_1:any = ((<any>this.context.mouseMoveEventHandler($event)) !== false);
      result = (pd_sub_1 && result);
    }
    if ((eventName == 'mouseup')) {
      const pd_sub_2:any = ((<any>this.context.mouseUpEventHandler($event)) !== false);
      result = (pd_sub_2 && result);
    }
    if ((eventName == 'touchstart')) {
      const pd_sub_3:any = ((<any>this.context.mouseDownEventHandler($event)) !== false);
      result = (pd_sub_3 && result);
    }
    if ((eventName == 'touchmove')) {
      const pd_sub_4:any = ((<any>this.context.mouseMoveEventHandler($event)) !== false);
      result = (pd_sub_4 && result);
    }
    if ((eventName == 'touchend')) {
      const pd_sub_5:any = ((<any>this.context.mouseUpEventHandler($event)) !== false);
      result = (pd_sub_5 && result);
    }
    if ((eventName == 'window:resize')) {
      const pd_sub_6:any = ((<any>this.context.resizeEventHandler($event)) !== false);
      result = (pd_sub_6 && result);
    }
    if ((eventName == 'document:mousemove')) {
      const pd_sub_7:any = ((<any>this.context.mouseMoveEventHandler($event)) !== false);
      result = (pd_sub_7 && result);
    }
    if ((eventName == 'document:mouseup')) {
      const pd_sub_8:any = ((<any>this.context.mouseUpEventHandler($event)) !== false);
      result = (pd_sub_8 && result);
    }
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean,emit3:boolean,emit4:boolean,emit5:boolean,emit6:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onDragStart.subscribe(_eventHandler.bind(view,'onDragStart'))); }
    if (emit1) { (this.subscription1 = this.context.onDrag.subscribe(_eventHandler.bind(view,'onDrag'))); }
    if (emit2) { (this.subscription2 = this.context.onDragStop.subscribe(_eventHandler.bind(view,'onDragStop'))); }
    if (emit3) { (this.subscription3 = this.context.onResizeStart.subscribe(_eventHandler.bind(view,'onResizeStart'))); }
    if (emit4) { (this.subscription4 = this.context.onResize.subscribe(_eventHandler.bind(view,'onResize'))); }
    if (emit5) { (this.subscription5 = this.context.onResizeStop.subscribe(_eventHandler.bind(view,'onResizeStop'))); }
    if (emit6) { (this.subscription6 = this.context.onItemChange.subscribe(_eventHandler.bind(view,'onItemChange'))); }
  }
}
